dev/Repeater/APITests/SelectionModelTests.cs

```diff
@@ -1134,6 +1134,83 @@ void ThrowIfRaisedSelectionChanged(SelectionModel sender, SelectionModelSelectio
             }
         }
 
+        [TestMethod] 
+        public void ValidateSelectionModeChangeFromMultipleToSingle()
+        {
+            RunOnUIThread.Execute(() =>
+            {
+                SelectionModel selectionModel = new SelectionModel();
+                selectionModel.Source = Enumerable.Range(0, 10).ToList();
+
+                // First test: switching from multiple to single selection mode with just one selected item
+                selectionModel.Select(4);
+
+                selectionModel.SingleSelect = true;
+
+                // Verify that the item at index 4 is still selected 
+                Verify.IsTrue(selectionModel.SelectedIndex.CompareTo(Path(4)) == 0, "Item at index 4 should have still been selected");
+
+                // Second test: this time switching from multiple to single selection mode with more than one selected item
+                selectionModel.SingleSelect = false;
+                selectionModel.Select(5);
+                selectionModel.Select(6);
+
+                // Now switch to single selection mode
+                selectionModel.SingleSelect = true;
+
+                // Verify that 
+                // - only one item is currently selected
+                // - the currently selected item was among the previously selected items in multiple selection mode
+                Verify.AreEqual(1, selectionModel.SelectedIndices.Count,
+                    "Exactly one item should have been selected now after we switched from Multiple to Single selection mode");
+                Verify.IsTrue(selectionModel.SelectedIndices[0].CompareTo(selectionModel.SelectedIndex) == 0);
+                Verify.IsTrue(
+                    selectionModel.SelectedIndex.CompareTo(Path(4)) == 0
+                    || selectionModel.SelectedIndex.CompareTo(Path(5)) == 0
+                    || selectionModel.SelectedIndex.CompareTo(Path(6)) == 0,
+                    "The currently selected item should have been one of the previously selected items");
```

Comment:
Do we really just want to select "one" of the items and not the first index? Should we try to not have undefined behavior here and e.g. always lowest or highest index when we switch?

Reply:
Should be the first (or lowest) index from the list of all selected indices. The indices are all sorted, so the that would be the first one.

Reply:
I wasn't under the impression that there is a public contract specifying which exact item will be picked when switching. This might be a problem with the available documentation on docs.microsoft.com though. As such, I felt the safe way here was for me to check if the now selected item was *any* of the previously selected items in Multiple selection mode.

Obviously, this should be updated if the exact selection behavior upon switch is actually part of a public API and we can test for an explicit element here.